hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/security/x509/certificate/client/DefaultCertificateClient.java

```diff
@@ -1060,4 +1065,37 @@ public long getLocalCrlId() {
   public void setLocalCrlId(long crlId) {
     this.localCrlId = crlId;
   }
+
+  @Override
+  public synchronized KeyStoresFactory getServerKeyStoresFactory()
+      throws CertificateException {
+    if (serverKeyStoresFactory == null) {
+      serverKeyStoresFactory = SecurityUtil.getServerKeyStoresFactory(
+          securityConfig, this, true);
+    }
+    return serverKeyStoresFactory;
+  }
+
+  @Override
+  public KeyStoresFactory getClientKeyStoresFactory(boolean mutualAuth)
+      throws CertificateException {
+    Preconditions.checkArgument(mutualAuth,
```

Comment:
If this precondition we want to keep for now, I would suggest to move this check to the method in SecurityUtil, and also perform the check for the serverkeystorefactory getter, or just simply hardwire there mutual auth without giving the option to choose via parameter until we extend support for non-mtls cases.

Reply:
Make sense.